packages/devices/origin-device-registry-irec-local-api/src/device/device.service.ts

```diff
@@ -49,20 +62,23 @@ export class DeviceService {
         return this.repository.find(options);
     }
 
-    async updateStatus(id: string, status: DeviceStatus): Promise<Device> {
+    async update(user: ILoggedInUser, id: string, deviceData: UpdateDeviceDTO): Promise<Device> {
         const device = await this.findOne(id);
 
         if (!device) {
             return null;
         }
 
-        await this.repository.update(device.id, { status });
-
-        this.eventBus.publish(new DeviceStatusChangedEvent(device, status));
+        const irecDevice = await this.irecDeviceService.update(user, device.code, deviceData);
```

Comment:
According to I-REC API documentation, editing is not allowed when `Updates can be made to devices with any status except withdrawn and approved.` I think we should handle the error and give the appropriate error, otherwise the users will get error 500.

Reply:
Thanks for noticing it. Actually, it allowed editing the device in all states except "InProgress"